internal/pkg/source/source.go

```diff
@@ -162,11 +162,35 @@ func (s *Source) referrersToVisit(n ssa.Node, maxInstrReached map[*ssa.BasicBloc
 			continue
 		}
 
+		if e, ok := r.(*ssa.Extract); ok && shouldNotVisitExtract(n, e) {
+			continue
+		}
+
 		referrers = append(referrers, r)
 	}
 	return referrers
 }
 
+// shouldNotVisitExtract determines if the extract is extracting a boolean from
+// a comma-ok map lookup or chan recv, in which case we should not visit it,
+// because a boolean cannot meaningfully be tainted.
```

Comment:
> because a boolean cannot meaningfully be tainted.

If this is our motivation, then do we care about the `Extract`?  Could we not just check for `bool` type and skip that irrespective of context?  I.e., considering this case:

```golang
func Foo(s Source) {
  var ok bool = s.IsReady()
  Sinkf("Source is ready: %v", ok)  // We would currently produce a report here, right?
  // ...
}
```

Even without an `Extract`, if we assert that a `bool` shouldn't carry taint, then we're missing this sort of case.

Reply:
I agree, I think the implementation for that will be simpler and it'll cover more cases.

My motivation for *not* banning all booleans was that it might allow us to discover more bugs in the future. That's a pretty weak motivation though.

I implemented the boolean check and added a few tests.